archlinux/arch-security-tracker
# 171 - add possibility to use SSO and create users just-in-time

As per our recent discussion on #171, I'd like to show you my POC for SSO and JIT user creation. For now, you can configure it via the default configuration file. I think there could be a better location for that, though.

What my implementation does:
- Configurable login provider: SSO or standard password/email login
- User creation if authentication happens via SSO with role extracted from a given group-role mapping
- User update with role according to the aforementioned group-role mapping

## How to test it
1. Run `docker run -p 8080:8080 -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin jboss/keycloak`
2. Log in to Keycloak using the admin:admin credentials
3. Create the arch-securiy-tracker client in Keycloak using the provided configuration file in `test/`
3. Create a group attribute mapper on the newly created client in Keycloak like so https://stackoverflow.com/questions/56362197/keycloak-oidc-retrieve-user-groups-attributes - it will attach all groups that a user is member of during login
4. Create users and groups as you see fit
5. Adapt the arch-security-tracker configuration in `config` with the OIDC secrets and appropriate group names
6. Start the application and log in/out until your browser tell you to stop :D
